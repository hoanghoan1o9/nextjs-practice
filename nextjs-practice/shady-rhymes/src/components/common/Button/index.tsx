// Libraries
import React, { memo } from 'react';
import {
  Button as ButtonChakra,
  ButtonProps as ButtonPropsChakra,
} from '@chakra-ui/react';

interface ButtonProps extends ButtonPropsChakra {
  disabled?: boolean;
  size?: 'default' | 'primary' | 'secondary';
  variant?: 'default' | 'primary' | 'secondary';
  label: string;
  className?: string;
  children?: React.ReactNode;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

const ButtonBase = ({
  isDisabled = false,
  children,
  label,
  ...props
}: ButtonProps) => {
  return (
    <ButtonChakra
      isDisabled={isDisabled}
      data-testid="testButton"
      {...props}
    >
      {label}
      {children}
    </ButtonChakra>
  );
};

export const Button = memo(ButtonBase);
