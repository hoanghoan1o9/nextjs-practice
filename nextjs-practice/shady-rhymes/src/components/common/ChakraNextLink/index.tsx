// Libraries
import React from 'react';
import Link, { LinkProps } from 'next/link';
import {
  Link as ChakraLink,
  LinkProps as ChakraLinkProps,
} from '@chakra-ui/react';

type ChakraLinkAndNextProps = ChakraLinkProps & LinkProps;

export const ChakraNextLink = React.forwardRef(
  ({ href, children, ...props }: ChakraLinkAndNextProps, ref: any) => {
    return (
      <Link href={href} passHref>
        <ChakraLink ref={ref} {...props}>
          {children}
        </ChakraLink>
      </Link>
    );
  },
);
