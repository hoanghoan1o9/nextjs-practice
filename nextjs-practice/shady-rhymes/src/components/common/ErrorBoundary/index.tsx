// Libraries
import React, { Component, ErrorInfo } from 'react';
import { Box, Flex, Link } from '@chakra-ui/react';
import NextLink from 'next/link';

// Constants
import { APP_ERRORS } from '@constants/errors';
import { APP_MESSAGES } from '@constants/messages';

interface State {
  error: Error | null;
  errorInfo: ErrorInfo | null;
}

interface Props {
  children: JSX.Element[] | JSX.Element;
}

export class ErrorBoundary extends Component<Props, State> {
  state: State = { error: null, errorInfo: null };

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    this.setState({
      error,
      errorInfo,
    });
  }

  render() {
    if (this.state.error) {
      return (
        <Box
          minH='100vh'
          display='flex'
          alignItems='center'
          justifyContent='center'
          px='10px'>
          <Flex
            flexDir='column'
            justifyContent='center'
            alignItems='center'>
            <h2>{APP_ERRORS.DEFAULT_ERROR_APIS}</h2><br/>
            <NextLink href='/'><Link color='primary' >{APP_MESSAGES.BACK_TO_HOME}</Link></NextLink>
          </Flex>
        </Box>
      );
    }

   return this.props.children;
  }
}
