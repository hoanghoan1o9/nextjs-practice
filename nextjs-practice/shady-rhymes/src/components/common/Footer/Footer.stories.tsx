// Libraries
import Storybook from '@storybook/react';

// Components
import { Footer } from './index';

export default {
  title: 'Components/Footer',
  component: Footer,
} as Storybook.ComponentMeta<typeof Footer>;

export const DefaultFooter = () => <Footer />;
