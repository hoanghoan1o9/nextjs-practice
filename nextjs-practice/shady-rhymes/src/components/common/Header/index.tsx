// Libraries
import { Heading } from '@chakra-ui/react';

// Components
import NavBar from '@components/common/NavBar';

// Constants
import { NAV_ITEMS } from '@constants/navigation';

export const Header = () => {
  return (
    <Heading className="container" py="20px" mx="20">
      <NavBar navItems={NAV_ITEMS} />
    </Heading>
  );
};
