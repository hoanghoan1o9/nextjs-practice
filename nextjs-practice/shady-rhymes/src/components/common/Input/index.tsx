// Libs
import React, { KeyboardEvent, ChangeEvent } from 'react';
import {
  Box,
  Icon,
  Text,
  InputGroup,
  InputProps,
  InputRightElement,
  Input as InputChakra,
} from '@chakra-ui/react';
import { FaSearch } from 'react-icons/fa';

interface FormInputProps extends InputProps {
  className?: string;
  placeholder: string;
  defaultValue?: string;
  rightContent?: string;
  message?: string;
  variant?: 'withAddon';
  size?: 'withAddon';
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
  onKeyDown?: (event: KeyboardEvent<HTMLInputElement>) => void;
}

export const Input = ({
  isDisabled = false,
  isRequired = true,
  className = '',
  isReadOnly = false,
  placeholder,
  autoFocus = true,
  message,
  ...props
}: FormInputProps) => {
  return (
    <Box pos="relative" h="100vh">
      <Text variant={'danger'} ml={"20px"}>{message}</Text>
      <InputGroup>
        <InputChakra
          isDisabled={isDisabled}
          isRequired={isRequired}
          className={className}
          isReadOnly={isReadOnly}
          autoFocus={autoFocus}
          variant="withAddon"
          placeholder={placeholder}
          {...props}
        />
        <InputRightElement
          pos="absolute"
          top="5"
          right="10"
          p="2"
          pointerEvents="none"
        >
          <Icon as={FaSearch} />
        </InputRightElement>
      </InputGroup>
    </Box>
  );
};
