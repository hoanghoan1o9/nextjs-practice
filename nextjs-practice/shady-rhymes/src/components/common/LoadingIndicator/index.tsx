// Libraries
import React, { memo } from 'react';
import { Flex, Spinner, SpinnerProps } from '@chakra-ui/react';

interface LoadingIndicatorProps extends SpinnerProps {
  className?: string;
}

const LoadingIndicatorBase = ({
  size = 'xl',
  ...props
}: LoadingIndicatorProps) => {
  return (
    <Flex
      minH="100vh"
      alignItems="center"
      justifyContent="center"
      w="100%"
      h="100%"
    >
      <Spinner size={size} data-testid="testLoader" {...props} />
    </Flex>
  );
};

export const LoadingIndicator = memo(LoadingIndicatorBase);
