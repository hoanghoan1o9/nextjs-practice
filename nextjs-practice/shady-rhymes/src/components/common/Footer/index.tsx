// Libs
import { ReactNode } from 'react';

// Components
import Logo from '@components/common/Logo';
import { ChakraNextLink } from '@components/common/ChakraNextLink';

// Constants
import { APP_IMAGE_URLS } from '@constants/images';

import { Box, Container, SimpleGrid, Stack, Text } from '@chakra-ui/react';

const ListHeader = ({ children }: { children: ReactNode }) => {
  return (
    <Text fontWeight="500" fontSize="medium" mb={2} color="#fadcd9">
      {children}
    </Text>
  );
};

export const Footer = () => (
  <Box
    bg="linear-gradient(275.58deg, #000000 0%, #4B4B4B 100%)"
    data-testid="TestFooter"
  >
    <Container as={Stack} maxW="6xl" py={10}>
      <SimpleGrid
        templateColumns={{ sm: '1fr 1fr', md: '2fr 1fr 1fr 1fr 1fr' }}
        spacing={8}
      >
        <Stack spacing={6}>
          <Box>
            <Logo
              alt="Footer logo"
              url={APP_IMAGE_URLS.FOOTER_LOGO}
              blurDataURL={APP_IMAGE_URLS.FOOTER_LOGO}
            />
          </Box>
        </Stack>
        <Stack align={'flex-start'} color="#ffffff">
          <ListHeader>Features</ListHeader>
          <ChakraNextLink href={'#'}>Our Features</ChakraNextLink>
          <ChakraNextLink href={'#'}>Pricing</ChakraNextLink>
          <ChakraNextLink href={'#'}>Affiliate Program</ChakraNextLink>
          <ChakraNextLink href={'#'}>Press Kit</ChakraNextLink>
        </Stack>
        <Stack align={'flex-start'} color="#ffffff">
          <ListHeader>Support</ListHeader>
          <ChakraNextLink href={'#'}>Account</ChakraNextLink>
          <ChakraNextLink href={'#'}>Help</ChakraNextLink>
          <ChakraNextLink href={'#'}>Contact Us</ChakraNextLink>
        </Stack>
        <Stack align={'flex-start'} color="#ffffff">
          <ListHeader>Legal</ListHeader>
          <ChakraNextLink href={'#'}>Privacy Policy</ChakraNextLink>
          <ChakraNextLink href={'#'}>Terms of Use</ChakraNextLink>
        </Stack>
        <Stack align={'flex-start'} color="#ffffff">
          <ListHeader>Social Media</ListHeader>
          <ChakraNextLink href={'#'}>Twitter</ChakraNextLink>
          <ChakraNextLink href={'#'}>Product Hunt</ChakraNextLink>
          <ChakraNextLink href={'#'}>Instagram</ChakraNextLink>
          <ChakraNextLink href={'#'}>Github</ChakraNextLink>
        </Stack>
      </SimpleGrid>
    </Container>
    <Box py={10}>
      <Text pt={6} fontSize="default" textAlign="center" color="#ffffff">
        Copyright 2021 Â©shady rhymes
      </Text>
    </Box>
  </Box>
);
