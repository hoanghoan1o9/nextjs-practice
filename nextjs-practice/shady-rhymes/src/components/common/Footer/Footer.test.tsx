// Libs
import { render } from '@testing-library/react';

// Components
import { Footer } from '@components/common/Footer';

describe('Footer render', () => {
  it('Should render Footer component', () => {
    const { getByTestId } = render(<Footer />);

    const footer = getByTestId('TestFooter');
    expect(footer).toBeTruthy();
    expect(footer).toMatchSnapshot();
  });
});
