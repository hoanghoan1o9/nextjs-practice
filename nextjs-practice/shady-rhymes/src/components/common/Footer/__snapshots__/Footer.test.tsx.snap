// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render Should render Footer component 1`] = `
<div
  class="css-1luty40"
  data-testid="TestFooter"
>
  <div
    class="chakra-stack chakra-container css-165xph2"
  >
    <div
      class="css-f7a9bo"
    >
      <div
        class="chakra-stack css-16voevi"
      >
        <div
          class="css-0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 120px; height: 35px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <img
              alt="Footer logo"
              data-nimg="fixed"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
      <div
        class="chakra-stack css-15s87bl"
      >
        <p
          class="chakra-text css-4vhhf7"
        >
          Features
        </p>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Our Features
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Pricing
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Affiliate Program
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Press Kit
        </a>
      </div>
      <div
        class="chakra-stack css-15s87bl"
      >
        <p
          class="chakra-text css-4vhhf7"
        >
          Support
        </p>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Account
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Help
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Contact Us
        </a>
      </div>
      <div
        class="chakra-stack css-15s87bl"
      >
        <p
          class="chakra-text css-4vhhf7"
        >
          Legal
        </p>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Privacy Policy
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Terms of Use
        </a>
      </div>
      <div
        class="chakra-stack css-15s87bl"
      >
        <p
          class="chakra-text css-4vhhf7"
        >
          Social Media
        </p>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Twitter
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Product Hunt
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Instagram
        </a>
        <a
          class="chakra-link css-0"
          href="/#"
        >
          Github
        </a>
      </div>
    </div>
  </div>
  <div
    class="css-9452eu"
  >
    <p
      class="chakra-text css-1qh67hf"
    >
      Copyright 2021 Â©shady rhymes
    </p>
  </div>
</div>
`;
